pkg/metrics/metrics.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"sigs.k8s.io/jobset/pkg/constants"
+)
+
+var (
+	FailedTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Subsystem: constants.JobSetSubsystemName,
+			Name:      "jobset_failed_total",
+			Help:      `The total number of failed JobSets`,
```

Comment:
If this is a cumulative total, what happens if the jobset controller restarts? Will the metric become inaccurate?

Reply:
Oh! I haven't thought about this question. It may be worth testing. However, based on the metrics from other projects, it seems that this question hasn't been considered either.

Reply:
I see, can you try testing it out?

Reply:
sure, but I have other things to do these days, so I am quite busy. I may have to try it on the weekend or next week.

Reply:
update: I'll test this scenario tomorrow and record the results :)